docs/state_machine.mmd

```diff
@@ -14,11 +14,11 @@ Running --> Updating
 Updating --> ClusterStarting
 DeployFailed --> Updating
 
-ClusterStarting --> Savepointing
+ClusterStarting --> Cancelling
 ClusterStarting -- Create fails --> DeployFailed
 
-Savepointing --> SubmittingJob
-Savepointing -- Savepoint fails --> Recovering
+Cancelling --> SubmittingJob
```

Comment:
@maghamravi Not a fan of this approach.

Ideally I would not change the existing statemachine as much as possible. If just cancel is needed without savepointing, my recommendation would be to introduce a new state, say "Cancelling" that is reached from Savepointing if savepoint is disabled in spec. I believe "Recovering" is also not associated with Cancelling.

Reply:
Honestly, I think we clubbed "Cancel" and "Savepoint" (two independent actions) into one called "Savepointing" which IMO should be "Cancelling". 
  
Dunno if it's just me(I think?),  but  when I see the state "Savepointing', I am under the impression that the operator is triggering a "Savepoint" on the job (https://ci.apache.org/projects/flink/flink-docs-stable/monitoring/rest_api.html#jobs-jobid-savepoints ) and not Cancelling it (with the **cancels** option)
 
If the consensus is to have "Cancel" be called from Savepointing in cases when "savepoint" is disabled,  though I disagree,  I am happy to commit!

Reply:
Agreed. There are two things here:

1) Better name for the Savepointing state.
2) Separating the intention of this PR to a separate phase/state.

I merely am recommending (2) now. We can always revisit (1) and update the state names later.

Reply:
@anandswaminathan Made the necessary changes!